server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -730,10 +732,19 @@ public Boolean isCcsMinimizeRoundtrips() {
          * @param clusterAlias The cluster alias as specified in the cluster collection
          * @return Cluster object associated with teh clusterAlias or null if not present
          */
-        public AtomicReference<Cluster> getCluster(String clusterAlias) {
+        public Cluster getCluster(String clusterAlias) {
             return clusterInfo.get(clusterAlias);
         }
 
+        /**
+         * @param clusterAlias key with which the specified value is associated
+         * @param remappingFunction function to transform the oldCluster to a newCluster
```

Comment:
Rather than "transform" I would use the word "swap", since the former implies that the Cluster object is mutable. Also we might want to give help to the implementers on the rules to follow when implementing the remappingFunction:

1. return a new Cluster obj to swap it for the existing one
2. return the original Cluster obj to keep the existing one (you decide not to swap)
3. never return null (it will remove the map entry)
4. throwing an Exception is OK (the original map entry will be retained), but is generally discouraged.

Reply:
Done, I also renamed the function to be `swapCluster` instead of `updateCluster` to remark the immutability. WDYT?